include/revng/Support/IntegerSerialization.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/Support/Endian.h"
+
+#include "revng/ADT/CompilationTime.h"
+#include "revng/ADT/Concepts.h"
+
+namespace detail {
+
+inline constexpr auto IntEndianness = llvm::support::little;
+inline constexpr auto Unaligned = llvm::support::unaligned;
+
+template<SpecializationOf<std::tuple> T>
+inline constexpr size_t computeSize() {
+  size_t Result = 0;
+  compile_time::repeat<std::tuple_size_v<T>>([&Result]<size_t I>() {
+    Result += sizeof(std::tuple_element_t<I, T>);
+  });
+  return Result;
+}
+
+template<typename... T>
+struct SerializedIntsSize {};
+
+template<isUintOrEnum... Args>
+struct SerializedIntsSize<Args...> {
+  static constexpr size_t Size = computeSize<std::tuple<Args...>>();
+};
+
+template<SpecializationOf<std::tuple> T>
+struct SerializedIntsSize<T> {
+  static constexpr size_t Size = computeSize<T>();
+};
```

Comment:
I think these can be turned into a

```c++
template<typename T>
constexpr size_t SerializedIntSize;


template<isUintOrEnum... Args>
constexpr size_t SerializedIntSize<Args> = ...;
```

Reply:
This:
```
template<typename T>
constexpr size_t SerializedIntSize;
```
Does not work, it needs to be initialized. Whereas the `struct` version will throw an compile-time error because the `Size` member is not present.

Reply:
Mh, I tested it, but not with clang. Oh well.